src/cmd/rmi.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright © 2019 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/containers/toolbox/pkg/podman"
+	"github.com/containers/toolbox/pkg/utils"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	rmiFlags struct {
+		deleteAll   bool
+		forceDelete bool
+	}
+)
+
+var rmiCmd = &cobra.Command{
+	Use:   "rmi",
+	Short: "Remove one or more toolbox images",
+	Run: func(cmd *cobra.Command, args []string) {
+		rmi(args)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(rmiCmd)
+
+	flags := rmiCmd.Flags()
+	flags.BoolVarP(&rmiFlags.deleteAll, "all", "a", false, "Remove all toolbox images")
+	flags.BoolVarP(&rmiFlags.forceDelete, "force", "f", false, "Force the removal of used toolbox images")
+}
+
+func rmi(args []string) {
+	if rmiFlags.deleteAll {
+		logrus.Info("Fetching images with label=com.github.debarshiray.toolbox=true")
+		args := []string{"--filter", "label=com.github.debarshiray.toolbox=true"}
+		Dimages, err := podman.GetImages(args...)
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		logrus.Info("Fetching images with label=com.github.containers.toolbox=true")
+		args = []string{"--filter", "label=com.github.containers.toolbox=true"}
+		Cimages, err := podman.GetImages(args...)
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		var idKey string
+		if podman.CheckVersion("1.8.2") < 0 {
+			idKey = "ID"
+		} else {
+			idKey = "id"
+		}
+		images := utils.JoinJSON(idKey, Dimages, Cimages)
+		for _, image := range images {
+			logrus.Infof("Deleting image %s", image[idKey].(string))
+			err = removeImage(image[idKey].(string))
+			if err != nil {
+				if errors.As(err, &podman.ErrHasChildren) {
+					logrus.Errorf("Image '%s' has dependent children", image[idKey])
+				}
+				if errors.As(err, &podman.ErrNonExistent) {
+					logrus.Errorf("Image '%s' does not exist", image[idKey].(string))
+				}
+				logrus.Error("Internal Podman error: %w", err)
+			}
+		}
+	} else {
+		if len(args) == 0 {
+			logrus.Fatal("Missing argument")
+		}
+
+		for _, imageID := range args {
+			// Check if the container exists
+			logrus.Infof("Inspecting image %s", imageID)
+			args := []string{"inspect", "--format", "json", "--type", "image", imageID}
+			output, err := podman.CmdOutput(args...)
+			if err != nil {
+				if errors.As(err, &podman.ErrInternal) {
+					logrus.Fatalf("Image '%s' does not exist", imageID)
```

Comment:
Even though this is going to be correct way many more times than not, we shouldn't be guessing like this. `podman inspect` doesn't use a specific exit code to denote missing images. We should just throw a generic error, unless we can find a way to exactly catch the *image not found* case.